code/datums/crew_objective.dm

```diff
@@ -307,11 +330,161 @@ ABSTRACT_TYPE(/datum/objective/crew/janitor)
 			return 1
 
-//	bartender
+#define DRINK_OBJ_COUNT 3
+ABSTRACT_TYPE(/datum/objective/crew/bartender)
+/datum/objective/crew/bartender
+	shotgun
+		explanation_text = "Don't lose your shotgun!"
+		check_completion()
+			if(owner.current && owner.current.check_contents_for(/obj/item/gun/kinetic/riotgun))
+				return TRUE
+			else
+				return FALSE
+	drinks
+		var/completed = FALSE
+		var/ids[DRINK_OBJ_COUNT]
+		New()
+			..()
+			var/list/cocktails = concrete_typesof(/datum/reagent/fooddrink/alcoholic)
+			var/list/blacklist = list(
+				/datum/reagent/fooddrink/alcoholic/bitters,
+				/datum/reagent/fooddrink/alcoholic/beer,
+				/datum/reagent/fooddrink/alcoholic/bojack,
+				/datum/reagent/fooddrink/alcoholic/bourbon,
+				/datum/reagent/fooddrink/alcoholic/champagne,
+				/datum/reagent/fooddrink/alcoholic/cider,
+				/datum/reagent/fooddrink/alcoholic/cocktail_grog,
+				/datum/reagent/fooddrink/alcoholic/curacao,
+				/datum/reagent/fooddrink/alcoholic/dbreath,
+				/datum/reagent/fooddrink/alcoholic/freeze,
+				/datum/reagent/fooddrink/alcoholic/gin,
+				/datum/reagent/fooddrink/alcoholic/mead,
+				/datum/reagent/fooddrink/alcoholic/moonshine,
+				/datum/reagent/fooddrink/alcoholic/ricewine,
+				/datum/reagent/fooddrink/alcoholic/rum,
+				/datum/reagent/fooddrink/alcoholic/tequila,
+				/datum/reagent/fooddrink/alcoholic/vermouth,
+				/datum/reagent/fooddrink/alcoholic/vodka,
+				/datum/reagent/fooddrink/alcoholic/wine,
+				/datum/reagent/fooddrink/alcoholic/wine/white
+			)
+			cocktails -= blacklist
+			var/list/names[DRINK_OBJ_COUNT]
+			for(var/i = 1, i <= DRINK_OBJ_COUNT, i++)
+				var/choiceType = pick(cocktails)
+				var/datum/reagent/fooddrink/instance =  new choiceType
+				names[i] = instance.name
+				ids[i] = instance.id
+			explanation_text = "Mix a "
+			for (var/ingredient in names)
+				if (ingredient != names[DRINK_OBJ_COUNT])
+					explanation_text += "[ingredient], "
+				else
+					explanation_text += "and [ingredient] "
+			explanation_text += "using your cocktail shaker."
+
+		check_completion()
+			return completed == 2**DRINK_OBJ_COUNT-1 //Uses bit flags
+
+#define CAKE_OBJ_COUNT 3
+#define PIZZA_OBJ_COUNT 3
+ABSTRACT_TYPE(/datum/objective/crew/chef)
+/datum/objective/crew/chef
+	var/list/blacklist = list(
+		/obj/item/reagent_containers/food/snacks/burger/humanburger,
+		/obj/item/reagent_containers/food/snacks/donut/custom/robust,
+		/obj/item/reagent_containers/food/snacks/ingredient/meat/humanmeat,
+		/obj/item/reagent_containers/food/snacks/ingredient/meat/mysterymeat/nugget/flock,
+		/obj/item/reagent_containers/food/snacks/ingredient/pepperoni,
+		/obj/item/reagent_containers/food/snacks/meatball,
+		/obj/item/reagent_containers/food/snacks/mushroom,
+		/obj/item/reagent_containers/food/snacks/pickle/trash,
+		/obj/item/reagent_containers/food/snacks/pizza/xmas,
+		/obj/item/reagent_containers/food/snacks/plant,
+		/obj/item/reagent_containers/food/snacks/plant/glowfruit/spawnable,
+	)
+	cake
```

Comment:
ideally these would not be relatively pathed like this.

Reply:
fixed